client/state/sites/domains/selectors.js

```diff
@@ -65,12 +65,14 @@ export function getDecoratedSiteDomains( state, siteId ) {
 		return {
 			...domain,
 
-			// Add registration moment from registrationDate
```

Comment:
You're mistaking the two assemblers - the Redux one (https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/domains/assembler.js) and the.. umm.. reflux (or whatever was hip back then) one (https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/assembler.js).

For the Redux one, we keep the raw dates, because that's what we want to store in the tree/offline/etc. But when fetching the domain, we want to decorate them with the `moment` versions, because those are more developer-friendly, plus we usually convert them to those anyway.

For the old assembler, it's the same idea as with the above decorator, but since we don't store it anywhere, we can expose the `moment` versions without the need for an explicit decorator. And I'm sure you agree that we don't need the raw versions of dates if we have the `moment` ones.